backend/dataall/cdkproxy/stacks/environment.py

```diff
@@ -555,33 +555,25 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
             )
 
         # Create or import SageMaker Studio domain if ML Studio enabled
-        if self._environment.mlStudiosEnabled:
+        self.existing_sagemaker_domain = SageMakerDomain(
```

Comment:
Minor. We could create SageMakerDomain first and then we could re-use it.

```
domain = SageMakerDomain(
     stack=self,
     id='SageMakerDomain',
     environment=self._environment
)
has_existing_domain = domain..check_existing_sagemaker_studio_domain()
.....
domain.create_sagemaker_domain_resources(sagemaker_principals=[default_role] + group_roles)

```

Reply:
good call, thanks!